pushpackage/checksum.go

```diff
@@ -0,0 +1,19 @@
+package pushpackage
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"io"
+)
+
+// copyAndChecksum calculates a checksum while writing to another output
+func copyAndChecksum(w io.Writer, r io.Reader) (checksum string, err error) {
```

Comment:
probably don't need to name the return variables here.

and can just return the result of hex.EncodeToString directly